src/logger.ts

```diff
@@ -1,5 +1,27 @@
-export interface ILogger {
-  info(...params: any): any;
-  debug(...params: any): any;
-  error(...params: any): any;
+import debug from "debug";
+
+export interface ILogger<IReturn = void> {
+  info(...params: any): IReturn;
+  debug(...params: any): IReturn;
+  error(...params: any): IReturn;
```

Comment:
any params :(

Reply:
```suggestion
  error(msg: string, ...context: unknown): IReturn;
```
instead of unknown you could also put generic

Reply:
Well, msg also won't work as it can be msg, message, log...

Reply:
I know ignoring types is bad, but we don't stand to gain anything by complicated types as we only care for the structure of info, debug, error

Reply:
lol, param name has nothing to do with anything, javascript params are positional not named. We are just ensuring that whatever logger is passed must have first param string type. This will avoid type errors in case of something like pino logger where only arg is object I think